examples/subplots_axes_and_figures/colorbar_placement.py

```diff
@@ -0,0 +1,52 @@
+"""
+=================
+Placing Colorbars
+=================
+
+Colorbars indicate the quantitative extent of image data.  Placing in
+a figure is non-trivial because room needs to be made for them.
+
+The simplest case is just attaching a colorbar to each axes:
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+
+fig, axs = plt.subplots(2, 2)
+cm = ['RdBu_r', 'viridis']
+for nn in range(2):
+    for mm in range(2):
+        ax = axs[mm, nn]
+        pcm = ax.pcolormesh(np.random.random((20, 20)) * (nn + 1), cmap=cm[nn])
+        fig.colorbar(pcm, ax=ax)
+plt.show()
+
+######################################################################
+# The first column has the same type of data in both rows, so it may
+# be desirable to combine the colorbar which we do by calling
+# `.Figure.colorbar` with a list of axes instead of a single axes.
+
+fig, axs = plt.subplots(2, 2)
+cm = ['RdBu_r', 'viridis']
+for nn in range(2):
+    for mm in range(2):
+        ax = axs[mm, nn]
+        pcm = ax.pcolormesh(np.random.random((20, 20)) * (nn + 1), cmap=cm[nn])
+    fig.colorbar(pcm, ax=axs[:, nn], shrink=0.6)
+plt.show()
+
+######################################################################
+# Relatively complicated colorbar layouts are possible using this
+# paradigm.  Note that this example works far better with
+# ``constrained_layout=True``
+
+fig, axs = plt.subplots(3, 3, constrained_layout=True)
+for ax in axs.flatten():
```

Comment:
I would use `ax.flat`. While `flatten()` reshapes the array, `flat` directly returns an iterator over the array, which is all we are needing here.